Dockerfile

```diff
@@ -2,10 +2,15 @@ FROM alpine
 ARG CLOUD_SDK_VERSION=254.0.0
 ENV CLOUD_SDK_VERSION=$CLOUD_SDK_VERSION
 
-RUN apk add --no-cache curl python python3 py-crcmod py-pip python-dev libffi-dev bash libc6-compat openssh-client openssl-dev git gnupg rsync coreutils gcc libc-dev make nodejs npm
-RUN npm install 8
-RUN npm install 10
-RUN npm install 12
+RUN apk add --no-cache curl python python3 py-crcmod py-pip python-dev libffi-dev bash libc6-compat openssh-client openssl-dev git gnupg rsync coreutils gcc libc-dev make npm ca-certificates ncurses g++ libgcc linux-headers grep util-linux binutils findutils
```

Comment:
Do we need all of these extra tools in the image, or do we need them just for installing NVM?

To optimise this we can install them temporarily by saving them to a virtual package, then removing the virtual package. 

```
RUN apk add --no-cache --virtual nvm-build-deps \
  ca-certificates ncurses g++ libgcc linux-headers grep util-linux binutils findutils && \
  curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.2/install.sh | bash && \
  apk remove --virtual nvm-build-deps
```

Additionally, by keeping everything in one docker instruction we keep the resulting image size down. As each instruction is a layer in the docker image, which means when something is added in one layer, the size increases, even if you remove it in a later layer.

We can further optimise this by reducing the total number of layers too, as well as removing anything we don't need once the image is built.